x-pack/plugins/maps/server/data_indexing/get_indexes_matching_pattern.ts

```diff
@@ -26,6 +26,18 @@ export async function getMatchingIndexes(
     const errorStatusCode = error.meta?.statusCode;
     if (errorStatusCode === 404) {
       return response.ok({ body: { success: true, matchingIndexes: [] } });
+    } else if (errorStatusCode === 401) {
+      return response.unauthorized({
```

Comment:
I don't think your route needs to go to this level of checking. The usage in the frontend is just throwing this information away. I think it would just be best to return a message with success: false and the error and avoid all of the error condition special handling. Keep simple things simple